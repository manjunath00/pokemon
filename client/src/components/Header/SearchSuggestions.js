import React, { useEffect, useRef } from "react";
import { Link } from "react-router-dom";

function SearchSuggestions({
  searchSuggestions, 
  setShowSearchResults,
}) { 
  const search = useRef(3);
  useEffect(() => {
    document.addEventListener("click", searchView); 

    return function cleanUp() {
      document.removeEventListener("click", searchView);
    };
  }, []);

  const searchView = (e) => {
    if (!search.current.contains(e.target)) {
      closeSuggestions();
    }
  };

  const closeSuggestions = () => {
    setShowSearchResults(false);
  };

  return (
    <ul
      className='search--suggestions'
      ref={search} >
      {searchSuggestions.length > 0
        ? searchSuggestions.map((item) => (
            <li key={item.pokemonName}>
              <Link
                to={`/pokemon/${item._id}`}
                onClick={() => closeSuggestions()}>
                {item.pokemonName}
              </Link>
            </li>
          ))
        : null}
    </ul>
  );
}

export default SearchSuggestions;
