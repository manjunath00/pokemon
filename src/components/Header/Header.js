import React from "react";
import { Link } from "react-router-dom";

function Header() {
  return (
    <header className='header'>
      <Link to='/' className='header__logo'>
        Pokedex
      </Link>
      <form>
        <input type='text' className='search-bar' />
      </form>
    </header>
  );
}

export default Header;
