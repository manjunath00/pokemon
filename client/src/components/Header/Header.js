import React, { useState } from "react";
import { Link } from "react-router-dom";
import SearchSuggestions from "./SearchSuggestions";

function Header({
  searchSuggestions,
  searchPokemon,
  searchTerm,
  onSearchSubmitFunc,
  setSearchTerm,
  reload,
  onReload,
}) {
  const [showSearchResults, setShowSearchResults] = useState(false);
  const [searchValue, setSearchValue] = useState("");

  const onCharEnter = (e) => {
    // console.log(e.target.value);
    const char = e.target.value.trim();
    setSearchValue(char);
    setSearchTerm(char);
    searchPokemon(char);
    setShowSearchResults(true);
  };

  const onSearchSubmit = (e) => {
    e.preventDefault();
    searchPokemon();
    onSearchSubmitFunc(searchValue);
    setShowSearchResults(false);
    // console.log("submitting", searchTerm);
  };

  const onGoBack = () => {
    onReload(!reload);
  };

  return (
    <>
      <header className='header'>
        <Link to='/' className='header__logo' onClick={() => onGoBack()}>
          Pokedex
        </Link>
        <div className='search'>
          <form className='search-bar' onSubmit={onSearchSubmit}>
            <input
              type='text'
              className='search-bar--input'
              value={searchValue}
              onChange={(e) => onCharEnter(e)}
            />
            <button type='submit' className='search-bar--submit'>
              <i className='fa fa-search' aria-hidden='true'></i>
            </button>
          </form>
          {showSearchResults ? (
            <SearchSuggestions
              searchSuggestions={searchSuggestions}
              setShowSearchResults={setShowSearchResults}
            />
          ) : null}
        </div>
      </header>
      <div className='new-button'>
        <Link className='btn-add' to='/new'>
          +
        </Link>
      </div>
    </>
  );
}

export default Header;
