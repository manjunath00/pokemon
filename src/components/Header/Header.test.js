import React from "react";
import { shallow } from "enzyme";

import Header from "./Header";

it("Renders static component with no props", () => {
  expect(shallow(<Header />)).toMatchSnapshot();
});
