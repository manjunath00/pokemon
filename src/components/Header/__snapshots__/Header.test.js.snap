// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders static component with no props 1`] = `ShallowWrapper {}`;
